src/main/java/org/dcsa/ebl/controller/TransportDocumentController.java

```diff
@@ -16,61 +19,71 @@
     }
 
-    @GetMapping
     @Override
-    public Flux<TransportDocumentTO> findAll(ServerHttpResponse response, ServerHttpRequest request) {
-        return super.findAll(response, request);
+    public String getType() {
+        return "TransportDocument";
     }
 
-    @GetMapping(path="{ID}")
-    @Override
-    public Mono<TransportDocumentTO> findById(@PathVariable UUID ID) {
-        return super.findById(ID);
+    @GetMapping
+    public Flux<TransportDocument> findAll(ServerHttpResponse response, ServerHttpRequest request) {
+        ExtendedRequest<TransportDocument> extendedRequest = new ExtendedRequest<>(extendedParameters,
+                TransportDocument.class);
+
+        try {
+            Map<String, String> params = request.getQueryParams().toSingleValueMap();
+            extendedRequest.parseParameter(params);
+        } catch (GetException e) {
+            return Flux.error(e);
+        }
+
+        return transportDocumentTOService.findAllExtended(extendedRequest).doOnComplete(
+                () -> extendedRequest.insertHeaders(response, request)
+        );
     }
 
-    @PutMapping( path = "{ID}")
-    @Override
-    public Mono<TransportDocumentTO> update(@PathVariable UUID ID, @Valid @RequestBody TransportDocumentTO transportDocumentTO) {
-        return super.update(ID, transportDocumentTO);
+    @GetMapping(path="{transportDocumentID}")
+    public Mono<TransportDocumentTO> findById(@PathVariable UUID transportDocumentID, @RequestParam(defaultValue = "false") boolean displayCharges) {
+        return transportDocumentTOService.findById(transportDocumentID, displayCharges);
     }
 
+    @PutMapping( path = "{transportDocumentID}")
+    public Mono<TransportDocumentTO> update(@PathVariable UUID transportDocumentID, @Valid @RequestBody TransportDocumentTO transportDocumentTO) {
+        return Mono.error(new UpdateException("Not possible to update a TransportDocument"));
+    }
 
-    @Override
+    @PostMapping
+    @ResponseStatus( HttpStatus.CREATED )
     public Mono<TransportDocumentTO> create(@Valid @RequestBody TransportDocumentTO transportDocumentTO) {
-        return Mono.error(new CreateException("Not possible to create a TransportDocument"));
+        return transportDocumentTOService.create(transportDocumentTO);
     }
 
     @DeleteMapping
     @ResponseStatus( HttpStatus.NO_CONTENT )
-    @Override
     public Mono<Void> delete(@RequestBody TransportDocumentTO transportDocumentTO) {
         return Mono.error(new DeleteException("Not possible to delete a TransportDocument"));
     }
 
-    @DeleteMapping( path ="{ID}" )
+    @DeleteMapping( path ="{transportDocumentID}" )
     @ResponseStatus( HttpStatus.NO_CONTENT )
-    @Override
-    public Mono<Void> deleteById(@PathVariable UUID ID) {
+    public Mono<Void> deleteById(@PathVariable UUID transportDocumentID) {
         return Mono.error(new DeleteException("Not possible to delete a TransportDocument"));
     }
```

Comment:
I would probably use 403 Forbidden as return code for these (DeleteException gives a 400 Bad request).